src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -251,10 +249,7 @@ private static IntPtr GetSslMethod(bool isServer, long options)
                 }
             }
 
-            if (IntPtr.Zero == method)
-            {
-                throw CreateSslException("Failed to get SSL method");
-            }
+            Debug.Assert(IntPtr.Zero != method, "Failed to get SSL method");
```

Comment:
In remote scenario if static cctor fails, this method will return null. As a result untimately innerContext will become invalid in allocateSSLContext & will throw the exception there.